### DISCUSSION TITLE
TypeError: cannot read property 'signUp'

### DISCUSSION THREAD

**Discussion by collectivec0de** [supabase/supabase#670] [Questions]
![Screen Shot 2021-02-03 at 9 45 04 PM](https://user-images.githubusercontent.com/39285071/106846035-5ba3e900-6669-11eb-8a6e-2bcb96ed1ff1.png)
Hi,

I'm trying to set up signup through integrating supabase with webflow. I'm using the signup code block from the supabase dashboard. I'm receiving the error: 

Uncaught (in promise) TypeError: Cannot read property 'signUp' of undefined
    at HTMLInputElement.signup
    
Here is the code:

<script>
  joinButton.addEventListener('click', signup);
  async function signup() {
    joinButton.style.display = 'none';
    joinError.style.display = 'none';
    var email = joinEmail.value;
    var password = joinPassword.value;
    let { user, error } = await supabase.auth.signUp({
      email: 'example@email.com',
      password: 'example-password'
    }).catch(function (error)
    {
      console.log(error);
    }) 
  };
</script>

**Comment by inian** [comment]
Hi @Shredcer10000000000, looks like the Supabase JS library has not been included in the page. Are you using the embed component to do it?

**Reply by collectivec0de** [reply]
Hey, thanks for the reply.

That definitely could be the case. How would I include it? supabase-{lang}?

**Reply by inian** [reply]
Yup, try including supabase-js. You can find more information in this thread - https://github.com/supabase/supabase/issues/218#issuecomment-738193724

I haven't tried it out myself, but let me know if you are able to successfully integrate with your Webflow project :)

**Reply by collectivec0de** [reply]
The import seems to be working and the code is able to recognize supabase, but it doesn't seem to recognize the signUp API.
Am I including it correctly in the source tag, like this?
<script src="https://unpkg.com/@supabase/supabase-js">

**Reply by inian** [reply]
Here is a working example with a Webflow project @Shredcer10000000000 https://github.com/supabase/supabase/issues/218#issuecomment-826520635
